docs/README.md

```diff
@@ -0,0 +1,208 @@
+
+#### Variations
+In an Institution environment, a Cloud Agent is less important since the Edge Agent is likely a persistent end point using on-premise or Cloud hosting. It may still be useful to have a Cloud Agent to prevent correlation. Regardless, an IEA without a Cloud Agent should still look to the rest of the world like any other Agent, so it should have endpoint message handling like that of a Cloud Agent - e.g. expect to be sent anon-crypt messages with a transport instructions and an encrypted payload.
+
+### Hubs
+A Hub is much like a Cloud Agent, but rather than focusing only on messaging (transport) as defined above for a Cloud Agent, the Hub also stores and shares data (potentially including Verifiable Credentials), on behalf of its owner. All of the data held by the Hub is en/decrypted by the Edge Agent, so it is the data that moves between the Edge and Hub, and not keys.  The Hub storage can (kind of) be thought of as a remote version of a Wallet without the keys, but is intended to hold more than just the Verifiable Credentials of an Edge Agent wallet. The idea is that the user can push lots of, for example, app-related data to the Hub, and a Service would be granted permission by the Owner to directly access the data without having to go to the Edge Agent. For example, a Hub-centric music service would store the owner’s config information and playlists on the Hub, and the Service would fetch the data from the Hub on use instead of storing it on it’s own servers.
+
+## Messaging Protocol
+### Transport Payload - Message Packaging/Unpackaging
+The process of handling messages should be consistent and independent of the transport mechanism of an agent implementation. The payload put onto the transport should be created in the following way:
+* Anon-encrypt the base message structure
+* [Base64](https://tools.ietf.org/html/rfc4648) encode the encrypted message
+
+The point in the process after the transport mechanism has delivered the payload should also be consistent and follow these steps:
+
+* [Base64](https://tools.ietf.org/html/rfc4648) decode the string
+* Anon-decrypt the bytes
+
+In order to anon encrypt/decrypt the payload, the endpoint (transport) mechanism MUST use a well known DID/VerKey associated with that endpoint called an Endpoint DID. No matter what kind of transport is used, the Endpoint DID should be used to package/unpackage the payload through/from the transport by following the steps outlined above.
```

Comment:
Secure transport of messages from source agent to destination agent is possible without a DID written to the ledger.
I'm willing to sit down and talk this over again.

Reply:
That's true. I'm using the phrase "well known" here to convey that you could either lookup the DID on the ledger, or share the Endpoint DIDs between two endpoints out of band. Either way, both agent endpoints need to "know" the Endpoint DID details in order to send messages back and forth.